File: Electronics_Shop_17.Services/Database/Coupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class Coupon : IHelperId, ISoftDelete
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public int DiscountAmount { get; set; }
        public int MinPurchaseAmount { get; set; }
        public int MaxUsagePerCustomer { get; set; }
        public int? ProductCategoryId { get; set; }//nepotrebno
        public int? AccessoryCategoryId { get; set; }//nepotrebno
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsActive { get; set; }

        //public virtual List<Order> Orders { get; set; }
        public bool isDeleted { get; set; }

    }
}

File: angular17_web/src/app/models/coupon/coupon.ts
export class Coupon {
  id!: number;
  code!: string;
  discountAmount!: number;
  minPurchaseAmount!: number;
  maxUsagePerCustomer!: number;
  productCategoryId?: number;
  accessoryCategoryId?: number;
  startDate!: Date;
  endDate!: Date;
  isActive!: boolean;
  isDeleted!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.code = data.code;
    this.discountAmount = data.discountAmount;
    this.minPurchaseAmount = data.minPurchaseAmount;
    this.maxUsagePerCustomer = data.maxUsagePerCustomer;
    this.productCategoryId = data.productCategoryId;
    this.accessoryCategoryId = data.accessoryCategoryId;
    this.startDate = data.startDate ? new Date(data.startDate) : undefined as any;
    this.endDate = data.endDate ? new Date(data.endDate) : undefined as any;
    this.isActive = data.isActive;
    this.isDeleted = data.isDeleted;
  }
}
